thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -484,44 +493,25 @@ export default Ember.Controller.extend({
   //
 
   actions: {
-    legendOnVisibility(urns, state) {
-      console.log('legendOnVisibility()');
-      const { invisibleUrns } = this.getProperties('invisibleUrns');
-      if (state) {
-        makeIterable(urns).forEach(urn => invisibleUrns.delete(urn));
-      } else {
-        makeIterable(urns).forEach(urn => invisibleUrns.add(urn));
-      }
-      console.log('legendOnVisibility: invisibleUrns', invisibleUrns);
-      this.set('invisibleUrns', new Set(invisibleUrns));
-    },
-
-    legendOnSelection(urns, state) {
-      console.log('legendOnSelection()');
+    onSelection(updates) {
+      console.log('onSelection()');
+      console.log('onSelection: updates', updates);
       const { selectedUrns } = this.getProperties('selectedUrns');
-      if (state) {
-        makeIterable(urns).forEach(urn => selectedUrns.add(urn));
-      } else {
-        makeIterable(urns).forEach(urn => selectedUrns.delete(urn));
-      }
-      console.log('legendOnVisibility: selectedUrns', selectedUrns);
+      Object.keys(updates).filter(urn => updates[urn]).forEach(urn => selectedUrns.add(urn));
+      Object.keys(updates).filter(urn => !updates[urn]).forEach(urn => selectedUrns.delete(urn));
```

Comment:
I think we can chain these two `.filter`:
`Object.keys(updates).filter(...).filter(...)`

Reply:
both lines perform different operations on different sets